version: '3.8'
name: 'pmanager'
services:
    server:
        container_name: pmanager_server
        build: .
        restart: on-failure
        depends_on:
            postgres:
                condition: service_healthy
        ports:
            - "8080:8080"
        environment:
            - EXTERNAL_PORT=8080
            - PGUSER=pmanroot
            - PGPASSWORD=pmanroot
            - PGDATABASE=pmanroot
            - PGHOST=pmanager_db
            - TOKEN_SECRET=xWmso9Os78GXhkruYB2FMWQPglostosEkCeyMxoWBo4d109RAdhG8pvdiBdJDls
        volumes:
            - ./app/src:/usr/src/app/src

    postgres:
        container_name: pmanager_db
        image: postgres:14.1-alpine
        restart: always
        environment:
            - POSTGRES_USER=pmanroot
            - POSTGRES_PASSWORD=pmanroot
            - POSTGRES_DB=pmanroot
        ports:
            - "5432:5432"
        healthcheck:
            test: ["CMD-SHELL", "sh -c 'pg_isready -U pmanroot -d pmanroot'"]
            interval: 5s
            timeout: 10s
            retries: 3

        volumes:
            - pmanager_data:/var/lib/postgresql/data

volumes:
    pmanager_data: {}
